module/VuFind/src/VuFind/View/Helper/Root/Record.php

```diff
@@ -711,4 +712,34 @@ protected function deduplicateLinks($links)
     {
         return array_values(array_unique($links, SORT_REGULAR));
     }
+
+    /**
+     * Get the source identifier + unique id of the record without spaces
+     *
+     * @param string $idPrefix Prefix for HTML ids
+     *
+     * @return string
+     */
+    public function getUniqueHtmlElementId($idPrefix = '')
+    {
+        return preg_replace(
+            "/\s+/",
+            '_',
+            ($idPrefix ? $idPrefix . '-' : '') . $this->getUniqueIdWithSourcePrefix()
+        );
+    }
+
+    /**
+     * Get the source identifier + unique id of the record
+     *
+     * @return string
+     */
+    public function getUniqueIdWithSourcePrefix()
+    {
+        if ($this->driver) {
+            return "{$this->driver->tryMethod('getSourceIdentifier')}"
+                . "|{$this->driver->tryMethod('getUniqueId')}";
```

Comment:
Both `getSourceIdentifier` and `getUniqueId` are required by the `AbstractBase` record driver, so I don't think it's necessary to use `tryMethod` here -- they should be guaranteed to exist in any legal record driver.